spec/factory_bot/attribute_list_spec.rb

```diff
@@ -59,28 +63,30 @@ def list(*attributes)
     attribute1 = double(:attribute1, name: :attribute1)
     attribute2 = double(:attribute2, name: :attribute2)
     attribute3 = double(:attribute3, name: :attribute3)
+    list = FactoryBot::AttributeList.new
 
-    subject.define_attribute(attribute1)
-    subject.apply_attributes(list(attribute2, attribute3))
-    expect(subject.to_a).to eq [attribute1, attribute2, attribute3]
+    list.define_attribute(attribute1)
+    list.apply_attributes(list(attribute2, attribute3))
+    expect(list.to_a).to eq [attribute1, attribute2, attribute3]
   end
 end
 
 describe FactoryBot::AttributeList, "#associations" do
-  let(:email_attribute) do
-    FactoryBot::Attribute::Dynamic.new(:email, false, ->(u) { "#{u.full_name}@example.com" })
-  end
-  let(:author_attribute)    { FactoryBot::Attribute::Association.new(:author, :user, {}) }
-  let(:profile_attribute)   { FactoryBot::Attribute::Association.new(:profile, :profile, {}) }
+  it "returns associations" do
+    email_attribute = FactoryBot::Attribute::Dynamic.new(
+      :email,
+      false,
+      ->(u) { "#{u.full_name}@example.com" },
+    )
+    author_attribute = FactoryBot::Attribute::Association.new(:author, :user, {})
+    profile_attribute = FactoryBot::Attribute::Association.new(:profile, :profile, {})
+    list = FactoryBot::AttributeList.new
```

Comment:
I think we could use the `build_attribute_list` function here and in the specs below as well:

```rb
list = build_attribute_list(
  email_attribute,
  author_attribute,
  profile_attribute,
)
```

Reply:
Fixed